### PR TITLE
feat: `count` endpoint

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

File: packages/core/src/indexing/sqlite/sqlite-index-api.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export class SqliteIndexApi implements DatabaseIndexApi {
       .insert({
         stream_id: args.streamID.toString(),
         controller_did: args.controller.toString(),
-        stream_content: args.streamContent.toString(),
+        stream_content: JSON.stringify(args.streamContent),
```
Comment (willex) [inline packages/core/src/indexing/sqlite/sqlite-index-api.ts:50]: Good catch!

**Review by willex — APPROVED** [review]
LGTM! 🏇

File: packages/http-client/src/remote-index-api.ts [hunk 2]
```diff
@@ -16,11 +16,15 @@ export class RemoteIndexApi implements IndexApi {
...
   constructor(apiUrl: URL) {
-    this._collectionURL = new URL('./collection', apiUrl)
+    this._collectionURL = new URL('./collection/', apiUrl)
   }
 
   async count(query: BaseQuery): Promise<number> {
-    throw new Error('Not implemented')
+    const queryURL = new URL('./count', this._collectionURL)
+    for (const key in query) {
+      queryURL.searchParams.set(key, query[key])
```
Comment (stbrody) [inline packages/http-client/src/remote-index-api.ts:25]: this won't actually serialize the filter right.  You'll need to use the `serializeObjectToSearchParams` function that I added in https://github.com/ceramicnetwork/js-ceramic/pull/2455

**Review by stbrody — COMMENTED** [review]
mostly looks good, only real comment is around using an object in the HTTP response instead of just the raw count scalar

File: packages/http-client/src/remote-index-api.ts (hunk 2 above)
Comment (ukstv) [reply packages/http-client/src/remote-index-api.ts:25]: Quite probable. As we discussed, you merge #2455, then I merge this one or rebase, so that we share the same function to handle the filters.

File: packages/http-client/src/__tests__/remote-index-api.test.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ test('model in query', async () => {
   ;(indexApi as any)._fetchJson = fauxFetch
   const result = await indexApi.query({ model: MODEL, first: 5 })
   expect(result).toEqual(EMPTY_RESPONSE)
-  expect(fauxFetch).toBeCalledWith(new URL(`https://example.com/collection?model=${MODEL}&first=5`))
+  expect(fauxFetch).toBeCalledWith(new URL(`https://example.com/collection/?model=${MODEL}&first=5`))
```
Comment (stbrody) [inline packages/http-client/src/__tests__/remote-index-api.test.ts:32]: this change seems surprising to me?  I'd expect the query params string to come directly on the `collection` endpoint, not after a slash to a new sub-endpoint?

File: packages/http-client/src/remote-index-api.ts [hunk 4]
```diff
@@ -17,19 +17,20 @@ export class RemoteIndexApi implements IndexApi {
   private readonly _collectionURL: URL
 
   constructor(apiUrl: URL) {
-    this._collectionURL = new URL('./collection', apiUrl)
+    this._collectionURL = new URL('./collection/', apiUrl)
```
Comment (stbrody) [inline packages/http-client/src/remote-index-api.ts:20]: why is this necessary?

File: packages/http-client/src/remote-index-api.ts [hunk 5]
```diff
@@ -17,19 +17,20 @@ export class RemoteIndexApi implements IndexApi {
   private readonly _collectionURL: URL
```
Comment (stbrody) [inline packages/http-client/src/remote-index-api.ts:17]: should we add a `_countURL` here so we don't need to construct it inline each time?

**Review by stbrody — COMMENTED** [review]
only question is why the GET query params come after a slash now instead of directly after the `collection` part of the URL.

File: packages/http-client/src/remote-index-api.ts (hunk 4 above)
Comment (ukstv) [reply packages/http-client/src/remote-index-api.ts]: For a server that does not matter. It matters for composing URLs. `./collection` means a "resource" named `collection` in the current directory of `apiURL`. If we want to have `./collection/count`, based on `this._collectionURL` using `new URL('./count', collectionURL)`, we have to indicate that `collection` is a "folder" not a "resource". Thus, a slash.

File: packages/http-client/src/remote-index-api.ts (hunk 5 above)
Comment (ukstv) [reply packages/http-client/src/remote-index-api.ts:17]: yup, to resolve the confusion

**Review by stbrody — APPROVED** [review]
LGTM!
